experimental/devices/as7262/as7262.go

```diff
@@ -35,42 +32,48 @@ var DefaultOpts = Opts{
 
 // New opens a handle to an AS7262 sensor.
 func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
-	ctx, cancel := context.WithCancel(context.Background())
-	cancel()
-	<-ctx.Done()
+	// The nil or zero values for gain and interrupt are valid
 	return &Dev{
 		c:         &i2c.Dev{Bus: bus, Addr: 0x49},
 		gain:      opts.Gain,
 		interrupt: opts.InterruptPin,
-		order:     binary.BigEndian,
+		done:      make(chan struct{}, 1),
 		timeout:   200 * time.Millisecond,
-		cancel:    cancel,
-		ctx:       ctx,
 	}, nil
 }
 
 // Dev is a handle to the as7262 sensor.
 type Dev struct {
-	mu        sync.Mutex
 	c         conn.Conn
 	timeout   time.Duration
 	interrupt gpio.PinIn
-	// A new context should be provided for long running operations.
-	cancel context.CancelFunc
-	ctx    context.Context
-	gain   Gain
-	order  binary.ByteOrder
+
+	// Mutable
+	mu   sync.Mutex
+	gain Gain
+	done chan struct{}
+
+	// Guards canceled
+	canceledMu sync.Mutex
+	canceled   bool
```

Comment:
Canceled is so that calls to Halt() do not send a done if a Halt() has already been called. I think canceled may not be the best name (cancelable is better). I think it is simpler to know that a done had been sent rather than the other way around ie: when a done had been received. In the case when there is something running like Sense() I can see how to respond that done was received, but starts to get a bit complicated if there is nothing to halt. Maybe I just need to think about it more, maybe read some more of context.Context.

Reply:
What about:
- Remove canceled, rename done to cancel.
- By default, d.cancel is nil
- Sense() does:
```
d.cancel := make(chan struct{})
d.cancelMu.Lock()
d.d.cancel = d.cancel
d.cancelMu.Unlock()
defer func() {
  d.cancelMu.Lock()
  d.d.cancel = nil
  d.cancelMu.Unlock()
}()
... uses d.cancel
```
- Halt does:
```
d.cancelMu.Lock()
if d.cancel != nil {
  select {
    case d.cancel <- struct{}{}:
    default:
  }
  d.cancel = nil
}
d.cancelMu.Unlock()
```